packages/venia-concept/src/components/MiniCart/miniCart.js

```diff
@@ -65,24 +76,56 @@ class MiniCart extends Component {
                     </Trigger>
                 </div>
                 <div className={classes.body}>
-                    <ProductList items={data} />
+                    {cartId ? (
+                        <ProductList
+                            currencyCode={cartCurrencyCode}
+                            items={cart.details.items}
+                        />
+                    ) : null}
                 </div>
                 <div className={classes.footer}>
                     <div className={classes.summary}>
-                        <dl className={classes.totals}>
-                            <dt className={classes.subtotalLabel}>
-                                <span>Subtotal (4 Items)</span>
-                            </dt>
-                            <dd className={classes.subtotalValue}>
-                                <Price currencyCode="USD" value={528} />
-                            </dd>
-                        </dl>
+                        {cartId && cart.totals && 'subtotal' in cart.totals ? (
+                            <dl className={classes.totals}>
+                                <dt className={classes.subtotalLabel}>
+                                    <span>
+                                        Subtotal{` (${
+                                            cart.details.items_qty
+                                        } Items)`}
+                                    </span>
+                                </dt>
+                                <dd className={classes.subtotalValue}>
+                                    <Price
+                                        currencyCode={cartCurrencyCode}
+                                        value={cart.totals.subtotal}
+                                    />
+                                </dd>
+                            </dl>
+                        ) : null}
                     </div>
+                    {checkout}
                 </div>
-                {checkout}
             </aside>
         );
     }
 }
 
-export default classify(defaultClasses)(MiniCart);
+export default compose(
+    classify(defaultClasses),
+    connect(
+        ({ cart }) => {
+            const details = cart && cart.details;
+            const cartId = details && details.id;
+            const cartCurrencyCode =
+                details &&
+                details.currency &&
+                details.currency.quote_currency_code;
+            return {
+                cart,
+                cartId,
+                cartCurrencyCode
+            };
+        },
+        { getCartDetails }
+    )
```

Comment:
Please consider defining mapStateToProps and mapDispatchToProps just before this export, as we've done in the other components. It's a bit less terse but more clear.